### PR TITLE
Add accepted_answers method and grey checkmark for as-yet unaccepted answers 

### LINKED ISSUE #1645: Add "accepted_answers" method to Node model, then add grey checkmark for as-yet unaccepted answers
### What happened just before the problem occurred
When looking at the new /wiki/hydrogen-sulfide, in a section including a table of things that had been tagged with hydrogen-sulfide, I saw one of my questions included there had been marked "(Answered)". However, I had not marked that question answered since the answer contributed was really more like comment material, and my question still stands. So, I'm concerned about it being so boldly marked as answered in the table. Would it be possible to only mark something as answered if the author has accepted an answer (rather than just an answer having been posted)? 

**Update: proposed solution (green/grey checkmark icons) added below with detail**

### Relevant URLs

https://publiclab.org/wiki/hydrogen-sulfide and https://publiclab.org/questions/gretchengehrke/08-23-2017/methods-for-or-experience-measuring-workplace-hydrogen-sulfide-gas

### PublicLab.org username

@gretchengehrke

### Browser, version, and operating system
Chrome, MacOS Sierra 10.12.6
screen shot 2017-09-26 at 10 51 46 am

For bug reports, fill out the above template; for feature requests, you can delete the template.

To learn how to write really great issues, which increases the chances they'll be resolved, see:

https://publiclab.org/wiki/developers#Contributing+for+non-coders

### PR DESCRIPTION
**Pull request by Diksha2008**
Fixes #1645 
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [X] all tests pass -- `rake test:all`
* [X] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [X] pull request is descriptively named with #number reference back to original issue

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.

Thanks!

### CODE REVIEW

**Comment by Diksha2008** [comment]
Sure, I'll work on it!

**Comment by jywarren** [comment]
Oh, super!

**Comment by Diksha2008** [comment]
@jywarren Since I am very new to this stuff, I think I have messed up very badly and even after googling a lot I am still clueless. Commits which are not made by me are showing up in this PR. Could you please help me out?

**Comment by jywarren** [comment]
Hi, no problem, were here to help. I can check in with some tips in an hour
or so, but until then maybe someone in the public lab chatroom at
publiclab.org/chat could help?

Thanks! Don't get discouraged!

File: test/unit/node_test.rb [hunk 1]
```diff
@@ -235,4 +235,13 @@ class NodeTest < ActiveSupport::TestCase
     assert !replaced
     assert_equal 'Jingle Jingle Bells', node.body
   end
+
+  test "question has an accepted answer" do
+    question2 = node(:question2)
+    assert_not question2.accepted_answers
```
Comment (jywarren) [inline test/unit/node_test.rb]: I think this is probably OK -- even with the merged in commits. I think the error is due to the `assert_not` here (see Dangerbot's feedback in the comments) and you may need to use `assert !question2.accepted_answers`. But does `accepted_answers` return a tru/false? i think you may want to `assert_equal 1, question2.accepted_answers.length` (not sure how many there are supposed to be). Make sense? 

Then you might accept a new answer and then ensure that the number of `accepted_answers` increased by 1.

File: test/unit/node_test.rb [hunk 2]
```diff
@@ -235,4 +235,13 @@ class NodeTest < ActiveSupport::TestCase
...
+
+    question = node(:question)
+    assert question.accepted_answers
```
Comment (jywarren) [inline test/unit/node_test.rb]: Likewise here, it probably won't return `true` but you could measure the length with `assert_equal 1, ______` for example.

Hope that helps!

File: test/unit/node_test.rb (hunk 1 above)
Comment (Diksha2008) [reply test/unit/node_test.rb]: `accepted_answer` method returns either true or false i.e. whether it has an accepted answer or not. So maybe I should change it to `assert !question2.accepted_answers` ?

**Comment by jywarren** [comment]
If you can make this has_ change is live to merge this in and call it completed!

**Comment by jywarren** [comment]
^ mistyped but you get what I mean :-)

**Comment by Diksha2008** [comment]
Sure! I'll change it!

**Comment by Diksha2008** [comment]
@jywarren I have changed `accepted_answers` method to `has_accepted_answers`. Please review it for any other change, if any

**Comment by jywarren** [comment]
Hi -- this looks perfect; merging now. If you're looking for a new challenge, please take a look at our `help-wanted` list: https://github.com/publiclab/plots2/labels/help-wanted

Thanks!

**Pull request merged**